libs/screen/image.cpp

```diff
@@ -1177,6 +1360,22 @@ void _fillTriangle(Image_ img, pxt::RefCollection *args) {
     );
 }
 
+//%
+void _fillPolygon4(Image_ img, pxt::RefCollection *args) {
```

Comment:
```suggestion
// This polygon fill is simular with fillTriangle(). Scan minY and maxY of all edges at each X, and draw a verticle line between (x,minY)~(x,maxY).
// Difference are: 
// Sort endpoints of each edge, x0<x1, to draw from left to right.
// But didn't sort edges, for it is too time consuming. Just call next(), and it will return immediatly if the x is not in range of the edge in horizon.
// NOTE, difference with triangles, edges of polygon can cross vertical line at a given X multi time, this algoritm can fill correctly only edges meet this codition: 
// Any vertical line(x) cross edges 2 times at most. 
// Fortunately, what ever perspective transfrom applied, a rectangle/trapezoid will be still met this condition.
// Ref: https://forum.makecode.com/t/new-3d-engine-help-filling-4-sided-polygons/18641/9?u=aqeeaqee
void _fillPolygon4(Image_ img, pxt::RefCollection *args) {
```